prosegrinder/__main__.py

```diff
@@ -5,22 +5,33 @@
 
 from prosegrinder.prose import Prose
 
+
 @click.command()
-@click.argument('file', required=True, type=click.File('r'))
-@click.option('-i', '--indent', required=False,
-    type=int, default=2, help="Python pretty-print json indent level.")
-@click.option('-s', '--save', required=False,
-    type=click.File('w'), help="File to save output to.")
-def cli(file, save, indent):
-    filename = click.format_filename(file.name)
-    text = file.read()
-    p = Prose(text)
-    d = {
-        "filename": filename,
-        "statistics": p.stats
-    }
-    j = json.dumps(d, indent=indent)
-    if (save):
-        save.write(j)
-    else:
-        click.echo(j)
+@click.argument("files", nargs=-1, required=True, type=click.File("r"))
+@click.option(
+    "-i",
+    "--indent",
+    required=False,
+    type=int,
+    default=2,
+    help="Python pretty-print json indent level.",
+)
+@click.option(
+    "-s",
+    "--save",
+    required=False,
+    type=click.File("w"),
+    help="File to save output to."
+)
+def cli(files, save, indent):
+    """setup the command line interface"""
+    for file in files:
```

Comment:
Output should be valid JSON. Make a list and append each file's stats to it, something like:
```python
l = []
for file in files:
    filename = click.format_filename(file.name)
    text = file.read()
    p = Prose(text)
    d = {"filename": filename, "statistics": p.stats}
j = json.dumps(l, indent=indent)
if save:
    save.write(j)
else:
    click.echo(j)
```

The `cli-default.json` will need updated, too.

Reply:
Good work, BTW!

Reply:
Good call. I updated it using your suggestion:
```python
def cli(files, save, indent):
    """Setup the command line interface"""
    processed_files = []
    for file in files:
        filename = click.format_filename(file.name)
        text = file.read()
        p = Prose(text)
        d = {"filename": filename, "statistics": p.stats}
        processed_files.append(d)
    j = json.dumps(processed_files, indent=indent)
    if save:
        save.write(j)
    else:
        click.echo(j)
```
This causes the output of `tests/resources/cli-default.json` to be an array and is valid JSON according to this [https://jsonlint.com/](url). 
I meant to add both changes as a single commit but goofed and forgot to make both updates simultaneously, so they are two separate commits. Also I tried to squash those other  intermediate commits in the commit history to clean it up, but couldn't figure out how to do that.